frontend/appflowy_tauri/src/appflowy_app/components/database/grid/GridCell/GridSelectCell/GridSelectCell.tsx

```diff
@@ -0,0 +1,172 @@
+  const [editingOption, setEditingOption] = useState<Database.SelectOption | undefined>();
+
+  const handleInput = useCallback((event: FormEvent) => {
+    const value = (event.target as HTMLInputElement).value;
+
+    setNewOptionName(value);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    setNewOptionName('');
+  }, []);
+
+  const handleChange = (event: SelectChangeEvent<string | string[]>) => {
+    const { target: { value } } = event;
+
+    const current = Array.isArray(value) ? value : [value];
+    const prev = cell?.data.selectOptions?.map(({ id }) => id);
+    const deleteOptionIds = prev?.filter(id => current.find(cur => cur === id) === undefined);
+
+    void service.updateSelectOptionCell(viewId, rowId, field.id, {
+      insertOptionIds: current,
+      deleteOptionIds,
+    });
+  };
+
+  const handleNewTagClick = async () => {
+    const exist = options.find(option => option.name.toLowerCase() === newOptionName.toLowerCase());
+
+    if (exist) {
+      return service.updateSelectOptionCell(viewId, rowId, field.id, {
+        insertOptionIds: [exist.id],
+      });
+    }
+
+    const option = await service.createSelectOption(viewId, field.id, newOptionName);
+
+    await service.insertOrUpdateSelectOption(viewId, field.id, [option], rowId);
+  };
+
+  const handleDetailsClick = useCallback((event: React.MouseEvent, option: Database.SelectOption) => {
+    event.stopPropagation();
+    setDomRef(event.target as HTMLElement);
+    setEditingOption(option);
+  }, []);
+
+  return (
+    <>
+      <Select
+        className="w-full"
+        classes={{
+          select: 'flex items-center gap-2 px-4 py-2 h-6',
+        }}
+        size="small"
+        value={selectedIds}
+        multiple={field.type === FieldType.MultiSelect}
+        input={<InputBase />}
+        IconComponent={() => null}
+        MenuProps={{
+          classes: {
+            list: 'py-5',
+          },
+          anchorOrigin: {
+            vertical: 'bottom',
+            horizontal: 'left',
+          },
+          transformOrigin: {
+            vertical: 'top',
+            horizontal: 'left',
+          },
+        }}
+        renderValue={(selected) => selected
+          .map((id) => options.find(option => option.id === id))
+          .map((option) => option && (
+            <Tag
+              key={option.id}
+              size="small"
+              color={option.color}
+              label={option.name}
+            />
+          ))}
+        onChange={handleChange}
+        onClose={handleClose}
+      >
+        <ListSubheader className="text-xs mb-4">
+          {t('grid.selectOption.createTag')}
+        </ListSubheader>
+        <ListSubheader className="flex">
+          <OutlinedInput
+            size="small"
+            value={newOptionName}
+            onInput={handleInput}
+          />
+        </ListSubheader>
+        {newOptionName ? (
+          <MenuItem
```

Comment:
```ts
const NewTagMenuItem = ({ newOptionName, selectedIds, handleNewTagClick, t }) => (
  newOptionName ? (
    <MenuItem
      className="mt-2 text-xs text-neutral-500 font-medium"
      value={selectedIds}
      onClick={handleNewTagClick}
    >
      {t('grid.selectOption.create')}
      <Tag className="ml-2" size="small" label={newOptionName} />
    </MenuItem>
  ) : null
);

const ExistingTagsHeader = ({ options, t }) => (
  options.length ? (
    <ListSubheader className="text-xs mt-4 mb-2">
      {t('grid.selectOption.orSelectOne')}
    </ListSubheader>
  ) : null
);

const ExistingTagMenuItem = ({ option, index, last, handleDetailsClick }) => (
  <MenuItem
    className={index !== last ? 'mb-2' : ''}
    key={option.id}
    value={option.id}
  >
    <div className="flex-1">
      <Tag key={option.id} size="small" color={option.color} label={option.name} />
    </div>
    <IconButton onClick={event => handleDetailsClick(event, option)}>
      <DetailsSvg className="text-base" />
    </IconButton>
  </MenuItem>
);

// Usage
<NewTagMenuItem newOptionName={newOptionName} selectedIds={selectedIds} handleNewTagClick={handleNewTagClick} t={t} />
<ExistingTagsHeader options={options} t={t} />
{options.map((option, index) => (
  <ExistingTagMenuItem option={option} index={index} last={options.length - 1} handleDetailsClick={handleDetailsClick} />
))}

```

Reply:
I tried modifying like you wrote, but ran into a problem.

The `Select` of `mui` has a limitation, it **must** use `MenuItem` as a direct child component to implement the selected logic.

So what I can do is keep the `MenuItem` component and wrap the children of `MenuItem` to another functional component called `SelectOptionItem`.